### DISCUSSION TITLE
Gitlab template uses deprecated json schema

### DISCUSSION THREAD

**Discussion by bgoareguer** [aquasecurity/trivy#4235] [Ideas]
## Description

The Gitlab report template ([contrib/gitlab.tpl](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab.tpl)) uses version [14.0.6](https://github.com/aquasecurity/trivy/blob/main/contrib/gitlab.tpl#L3) of the json schema for Gitlab container scanning reports.

This version of the json schema is deprecated. Gitlab suggests to use version 15.0.0 or higher.

![image](https://user-images.githubusercontent.com/43874676/231186347-0c40e757-3268-44c3-a991-f96af18fdd01.png)

**Comment by knqyf263** [comment]
It's been a while GitLab 15.0 including [free Trivy integration](https://gitlab.com/groups/gitlab-org/-/epics/2233) was released. I think it is a good time to deprecate `gitlab.tpl`.

**Comment by pismy** [comment]
Hey @knqyf263
GitLab's integration of Trivy scanners also comes with some discutable architectural choices (Auto-DevOps template, with a given Git worflow, ...).
Many projects and companies are also happy to integrate Trivy with their own policy, but still taking advantage of Trivy report integration in GitLab (that requires using this template)

Noticeable: two open-source projects are actually using this template directly:
* [to-be-continuous Docker template](https://gitlab.com/to-be-continuous/docker/#docker-trivy-job)
* [R2Devops Trivy template](https://r2devops.io/marketplace/gitlab/r2devops/hub/trivy_image)

**Comment by knqyf263** [comment]
Trivy now has a lot of features such as image scanning, misconfiguration scanning, Kubernetes cluster scanning, AWS account scanning, etc. It is hard to maintain the GitLab template as we don't have enough resources. Also, we don't use GitLab at the moment. We hope the community using GitLab maintains that.

**Comment by dklimpel** [comment]
If someone looking for a valid template for Gitlab 16. -> https://github.com/aquasecurity/trivy/pull/4035#issuecomment-1829306438
